drivers/sensor/si7006/si7006.c

```diff
@@ -1,35 +1,30 @@
 /*
  * Copyright (c) 2019 Electronut Labs
+ * Copyright (c) 2023 Trent Piepho <tpiepho@gmail.com>
  *
  * SPDX-License-Identifier: Apache-2.0
  */
 
-#define DT_DRV_COMPAT silabs_si7006
```

Comment:
Could you extract the generic part of this driver, which would be used by the si7006 and the sht21 drivers? Having it all in one driver named `si7006` makes it confusing.

Reply:
There is just one driver.  The only difference to support both sensors is literally two bits in one byte.

The driver should probably be renamed sht21, since that's the original device.  I can do that in a followup PR.  I find it's much better to do mass-renames in a PR of their own, not combined with a PR that also changes the code in other ways.

Reply:
You need to define `DT_DRV_COMPAT` somewhere for the DT_INST macros below to work. I'd expect to see something like this to support multiple compatibles in the driver: https://github.com/zephyrproject-rtos/zephyr/blob/600e7b7f21cfd179609372bb4f5e3bc5d33d19c2/drivers/sensor/tmp108/tmp108.c#L406-L414

Reply:
I assure you, this driver works.

I don't use the DT_INST macros.  The non "INST" versions work fine.  This seems nicer than needing to undef and redefine DT_DRV_COMPAT between each call.

Reply:
I thought more about splitting the driver.  It doesn't make sense.  tmp108 + as6212 show an example of one driver for two compatible devices.  This is like that.

Splitting it would mean putting everything into the common file.  Then making various static and private data global, i.e. the api definition object, the type for the data and config structs, the init function.  Zephyr doesn't use LTO, so there is a cost to making things non-static.  Then shc21.c and si7006.c become just one line each:
```c
DT_FOREACH_STATUS_OKAY_VARGS(silabs_si7006, SI7006_DEFINE, SI7006_READ_OLD_TEMP);
```
And likewise for sht21.

It's apparent that this just a convoluted way to add more source files solely for the purpose of adding more source files.

And your original issue, that the code for different-but-compatible sensors is in a file that uses the name si7006, is not improved in any way.  All the code (less one line) is still in one file with one name.  

It would be reasonable to mass rename si7006 to something else.  I think sht21, being the original, would be more correct.  Or another name.

Reply:
> I don't use the DT_INST macros. The non "INST" versions work fine. This seems nicer than needing to undef and redefine DT_DRV_COMPAT between each call.

Please keep the INST versions to be consistent with other drivers

Reply:
You can see the same method being used to change a driver from one device to multiple device here, https://github.com/zephyrproject-rtos/zephyr/pull/48042

It seems the existing policy is that method is just fine and no one had any problems with it.

Reply:
There isn't a policy one way or the other for the small number of drivers that support more than one compatible. The vast majority of drivers overall use DT_INST macros and diverging from that pattern makes them harder to review.

Reply:
You demanding that I do something differently, which was acceptable when other authors did it, doesn't feel fair and equitable to me.  If this is important, it would be important for other drivers to do it the same way too.  If it's not important, then why the need to demand it from me?

Reply:
Hi @xyzzy42, let me try to unstale the conversation here: coding conventions in the repository are global for few things that are well defined in the coding style, but everything else is then down to the subsystem maintainer and collaborators, and ultimately to the code author. It may feel wrong that the author is the lower priority there but the reality is that once the code is accepted, the maintainer is the one responsible for it in the not so uncommon case that the original author becomes unresponsive for whatever reason, so the code has to feel familiar to them.

This case, handling of mutliple compatibles, does not have a well defined pattern in the coding style. If the maintainer and collaborators have no preferences you can pick your style, but in this case they do, so the right thing to do is to follow their preferred style - nothing personal, no hard feelings, they would have asked it to other authors as well if they were submitting code to the subystem they are responsible for.